### PR TITLE
Ellie and Anika Lab 3

### PR DESCRIPTION
**Pull request by elliegibbs**

### CODE REVIEW

**Comment by elliegibbs** [comment]
Hi Kevin, Do you mind taking a look at step 8 for us and letting us know if we should keep writing the nwalign testset or not? It is getting an error instead of failing. How much detail are you looking for?
Thanks,
Ellie and Anika

File: Project.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name = "BISC195Labs"
 uuid = "3c894fce-1a5f-4a14-a547-781e8c68175f"
-authors = ["Kevin Bonham, PhD"]
+authors = ["Kevin Bonham, PhD", "Ellie Gibbs and Anika Luo"]
```
Comment (kescobo) [inline Project.toml]: You can each be listed separately, `... "Ellie Gibbs", "Anika Luo"]`

File: src/needleman_wunch.jl [hunk 2]
```diff
@@ -1,15 +1,56 @@
 # Labs 3 & 4
 
 """
-    nwscore(base1, base2)
+    nwscore(base1, base2; mismatch = -1, match = 1, gap = -1 )
 
 Your documentation here.
+
+This function compares nucleotides for genomic sequences
+    to see if they are matching(1), mismatching(-1), or gaps(-1).
```
Comment (kescobo) [inline src/needleman_wunch.jl]: what's with the weird indexing? Can we get this stuff better aligned?

File: src/needleman_wunch.jl [hunk 3]
```diff
@@ -1,15 +1,56 @@
...
+                [2] top-level scope
+                    @ REPL[81]:1
+
 """
-function nwscore(base1::Char, base2::Char)
-    
+function nwscore(base1::Char, base2::Char; match = 1, mismatch = -1, gap = -1)
+    if base1 == base2
+        return match
+    elseif base1 != base2
```
Comment (kescobo) [inline src/needleman_wunch.jl]: by definition, if we get to this branch, `base1 != base2`. So this can just be `else`

File: test/runtests.jl [hunk 4]
```diff
@@ -25,4 +25,13 @@ using Test
...
 
+    # Test double gap
+        @test_throws ArgumentError nwscore(nothing, nothing)
+
+end
+
+@testset "nwalign" begin
+    seq1="GCATGCU"
+    seq2="GATTACA"
+    @test any(c-> in(c, ["GCATG-CU","G-ATTACA","GCA-TGCU","G-ATTACA", "GCAT-GCU", "G-ATTACA"]), nwalign(seq1,seq2))
```
Comment (kescobo) [inline test/runtests.jl]: So, you're expecting to return a single string? Will it represent `seq1` or `seq2`? 

Are you planning to return the score? What about the scoring matrix?

**Review by kescobo — CHANGES_REQUESTED** [review]
This is looking pretty good! I'd think a little bit more about what you expect `nwalign()` to look like, but overall this is fine. You can put the `nwalign` tests into a different file or comment them out so that they don't bork your test runs

**Review by kescobo — APPROVED** [review]

**Pull request closed without merging**